src/components/Common/PollingProgressBar.tsx

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2023 Cortex Applications, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { LinearGauge } from '@backstage/core-components';
+import React, { useState } from 'react';
+import { ExponentialBackoff } from '../../utils/MathUtils';
+import { useInterval } from '../../utils/HookUtils';
+
+interface PollingProgressBarProps {
+  backoffMultiplier?: number;
+  done?: boolean;
+  initialDelay?: number;
+  maxDelay?: number;
+  poll: () => Promise<void>;
+  value: number;
+}
+
+export const PollingProgressBar: React.FC<PollingProgressBarProps> = ({
```

Comment:
nit: no need to export since you have the default export at the bottom

```suggestion
const PollingProgressBar: React.FC<PollingProgressBarProps> = ({
```